lib/invoker.js

```diff
@@ -24,21 +25,53 @@ const invoke = (userFunction) => {
     };
 };
 
-module.exports = (userFunctionUri, port) => {
-    const userFunction = lookUpFunction(userFunctionUri);
-
-    const server = new grpc.Server();
-    server.addService(services.RiffService, {invoke: invoke(userFunction)});
+const executeInit = async (initHook, timeout, server) => {
+    logger('Calling $init hook (if any)');
+    return await guardWithTimeout(initHook, timeout)
+        .then(() => logger('$init hook successfully completed (or skipped if none)'))
+        .catch((err) => {
+            logger(`Error during $init hook execution: ${err.toString()}`);
+            logger('Shutting down now');
+            server.forceShutdown();
```

Comment:
what is being shutdown? the server shouldn't be running

Reply:
in the current PR, the shutdown hook is run before the server shutdown. The hook could fail, in which case we force the server shutdown.
As you suggested above, though, the hook should be called after the server shutdown, so this code will disappear.